src/state/routing/types.ts

```diff
@@ -84,17 +90,28 @@ export interface QuoteResult {
   routeString: string
 }
 
-export const INITIALIZED = 'Initialized'
-export const NO_ROUTE = 'No Route'
-
-export type GetQuoteError = typeof INITIALIZED | typeof NO_ROUTE
-
-export type TradeResult = {
-  trade?: InterfaceTrade
-  gasUseEstimateUSD?: string
-  blockNumber: string
-}
+export type QuoteResult =
+  | {
+      state: QuoteState.INITIALIZED | QuoteState.NOT_FOUND
+      data?: QuoteData
+    }
+  | {
+      state: QuoteState.SUCCESS
+      data: QuoteData
+    }
 
-export type TradeQuoteResult = TradeResult | GetQuoteError
+export type TradeResult =
+  | {
+      state: QuoteState.INITIALIZED | QuoteState.NOT_FOUND
```

Comment:
it looks like `state: QuoteState.NOT_FOUND` is only coupled with the rest of the varaibles being undefined, so i would narrow down the type definition of `TradeResult` even further:

```
 {
   state: QuoteState.INITIALIZED | QuoteState.NOT_FOUND
   trade: undefined 
   gasUseEstimateUSD: undefined 
   blockNumber: undefined
  
```

Reply:
If I type them as `undefined` then I'm required to define them (as `undefined` 🤮 ).
So I'm going to narrow them to optional `undefined`, so I can continue to actually leave them undefined.

Reply:
That's fine!